src/sample-executors/BaseExecutor.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {Owned} from "solmate/src/auth/Owned.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+
+import {ResolvedOrder, SignedOrder} from "../base/ReactorStructs.sol";
+import {IReactorCallback} from "../interfaces/IReactorCallback.sol";
+import {IReactor} from "../interfaces/IReactor.sol";
+
+library Commands {
```

Comment:
ohhh interesting, reason why you decided to go with UR style commands vs a simple self delegatecall multisig? like https://github.com/Uniswap/v3-periphery/blob/main/contracts/base/Multicall.sol

worried constructing these commands may be a pain for backend code compared to just encoding array of calldata

Reply:
I guess some gas benefit this way?

Reply:
Also what if inheritors of this base want to add more commands? they can't edit the base commands type. If using a simple multicall they can just add new external fns which can be multicalled...

IMO this feels a bit too hardcore for a sample executor, would prefer a simple multicall

Reply:
Can check out Sara's old PR here which does this https://github.com/Uniswap/UniswapX/pull/115/files

Reply:
Agree that a generic multicall would be a much easier lift